run_project_tests.py

```diff
@@ -998,9 +993,7 @@ def have_working_compiler(lang: str, use_tmp: bool) -> bool:
     return True
 
 def have_java() -> bool:
-    if shutil.which('javac') and shutil.which('java'):
-        return True
-    return False
+    return bool(shutil.which('javac') and shutil.which('java'))
```

Comment:
Do we actually need `bool()` here?

Reply:
Yes, we do because the function type hint is a promise that we will return a `bool`.

Without `bool()` we would return a `str` on my Mac.

Reply:
Yeah, you're right.

Reply:
the `x and y` operator returns either x, or y, depending on whether x evaluates true. Same with `or`.

The `not` operator does convert to a guaranteed boolean.